text/0143-versioning-and-stability.md

```diff
@@ -0,0 +1,128 @@
+# OpenTelemetry versioning and stability
+
+OpenTelemetry is a large project with strict compatibility requirements. This proposal defines the stability gurantees offered by OpenTelemetry, along with a versioning and lifecycle proposal which defines how we meet those requirements.
+
+Language implementations are expected to follow this proposal exactly, unless a langague or package manager convention interferes significantly. Implementations must take this cross-language proposal, and produce a language-specific proposal which details how these requirements will be met.
+
+## Design goals
+
+**Provide end users with coherent, tested releases.**  
+We do not want users to run untested combinations of core packages. Users must be able to say “I am using OpenTelemetry v1.2.0 for Python” and have a shared understanding as to what that means.
+
+**Ensure that end users stay up to date with the latest release.**  
+We want all users to  stay up to date with the latest version of OpenTelemetry. We do not want to create hard breaks in support, of any kind, which leave users stranded on older versions. It must always be possible to upgrade to the latest minor version of OpenTelemetry, without creating a compilation error.
+
+**Never create a dependency conflict between packages which rely on different versions of OpenTelemetry. Avoid breaking all stable public APIs.**  
+Backwards compatibility is a strict requirement. Instrumentation APIs cannot create a version conflict, ever. Otherwise, OpenTelemetry cannot be embedded in widely shared libraries, such as web frameworks. Theoretically, APIs can be deprecated and eventually removed, but this is a process measured in years.
+
+**Allow for multiple levels of package stability within the same release.**  
+Provide maintainers a clear process for developing new, experimental APIs alongside stable APIs. DIfferent packages within the same release will have different levels of stability.
+
+## Relevant architecure
+
+OpenTelemetry is split into signals. Each signal provides a specialized form of observability.
+
+Each signal consists of four types of packages:
+
+**API -** Public interfaces depended upon by 3rd-party libraries and application code instrumented with  OpenTelemetry. Every signal has its own, independent API package. These individual APIs are then bundled up into a Global API for convenience.
+
```

Comment:
Global API package?

Reply:
This needs some more detail, we don't have a term for this in the spec right now. But, in general, it looks like languages follow a pattern: implement each signal separately, in their own packages. Then, implementations combine expose access to all of the combined signals in another package. Starting to name that package the "Global API," as it is a reasonable design but has implications for versioning.